### ISSUE TITLE
NaN results using backprop.py

### ISSUE THREAD

**Issue by tiagotrigger** [pybrain/pybrain#98]
I'm using a neuralnet to learn time series and when I use the backpropagation training all the results are NaN.
I use a FeedFoward (in:4, 5,5, out:1) net with TanhLayer and FullConnection between them.

This is the output warning that is shown:

/usr/local/lib/python2.7/dist-packages/PyBrain-0.3.1-py2.7.egg/pybrain/supervised/trainers/backprop.py:97: RuntimeWarning: overflow encountered in square
  error += 0.5 \* sum(outerr *\* 2)
/usr/local/lib/python2.7/dist-packages/PyBrain-0.3.1-py2.7.egg/pybrain/structure/modules/tanhlayer.py:15: RuntimeWarning: invalid value encountered in multiply
  inerr[:] = (1 - outbuf**2) \* outerr

This is the result of backpropTrainer.testondata():

All errors: [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]
Average error: nan
('Max error:', nan, 'Median error:', nan)

**Comment by RunshengSong** [comment]
I had the same problem, make sure your training data and training target have the same dimension. 
![image](https://cloud.githubusercontent.com/assets/9614963/9942177/91a87a1a-5d2e-11e5-887a-38907aee5f60.png)

**Comment by AveryLiu** [comment]
@c0d3rman Making the necessary attributes into np.float64 helped. I had the same problem and solved it by firstly normalizing all the input data(including the training target) and then converting all the numpy array used by the trainer into np.float64 type.

**Comment by SchmidtQQ** [comment]
Has anybody tried adding bias units to the network? It seems to mitigate the problem
